internal/highlight/highlight.go

```diff
@@ -325,6 +326,19 @@ func preSpansToTable(h string) (string, error) {
 		codeTd.AppendChild(codeCell)
 		codeTd.Attr = append(codeCell.Attr, html.Attribute{Key: "class", Val: "code"})
 	}
+	addNewRowsForTextNode := func(textNode *html.Node) {
+		// Text node, create a new table row for each newline at the end.
+		nodeData := textNode.Data
+		// Remove preceding newlines, if the entire node is *not* made up of newlines.
+		// This prevents us from counting the preceding newlines and appending them as rows at the end.
+		if len(nodeData) != strings.Count(nodeData, "\n") {
+			nodeData = strings.TrimLeft(nodeData, "\n")
+		}
+		newlines := strings.Count(nodeData, "\n")
+		for i := 0; i < newlines; i++ {
+			newRow()
+		}
+	}
```

Comment:
It would be nice if we didn't have to call `strings.Count` twice for each call of `addNewRowsForTextNode`.  Here is one way one could do it, though admittedly it is harder to read. What do you think? Maybe you find a better way?

```suggestion
addNewRowsForTextNode := func(textNode *html.Node) {
		// Text node, create a new table row for each newline at the end.
		nodeData := textNode.Data
		newline := strings.Count(nodeData, "\n")
		defer func() {
			for i := 0; i < newline; i++ {
				newRow()
			}
		}()
		if len(nodeData) == newline {
			return
		}
		// Remove preceding newlines, if the entire node is *not* made up of newlines.
		// This prevents us from counting the preceding newlines and appending them as rows at the end.
		newline = newline - (len(nodeData) - len(strings.TrimLeft(nodeData, "\n")))
	}
```

Reply:
Thanks for the suggestion, I rewrote the function to count and trim the newlines in the string only once. You can check again, let me know if it's ok.